homeassistant/components/hikvision/binary_sensor.py

```diff
@@ -227,7 +227,10 @@ async def async_added_to_hass(self) -> None:
         # Register callback with pyhik
         self._camera.add_update_callback(self._update_callback, self._callback_id)
 
-    @callback
     def _update_callback(self, msg: str) -> None:
-        """Update the sensor's state when callback is triggered."""
-        self.async_write_ha_state()
+        """Update the sensor's state when callback is triggered.
+
+        This is called from pyhik's event stream thread, so we must schedule
+        the state update to run in the event loop thread.
+        """
+        self.hass.loop.call_soon_threadsafe(self.async_write_ha_state)
```

Comment:
You can use the entity function schedule_update_ha_state that works threadsafe here.